### DISCUSSION TITLE
vc-drawings绘图组件中before-load和ready执行先后顺序问题求助

### DISCUSSION THREAD

**Discussion by xiaobaicxk** [zouyaoji/vue-cesium#447] [Q&A] [answered]
![image](https://github.com/zouyaoji/vue-cesium/assets/103714416/38bcd864-f161-4a24-b6b9-1d11db42b453)
我设置了before-load和reday方法
![image](https://github.com/zouyaoji/vue-cesium/assets/103714416/3328e205-d504-44a9-a437-a29cec3a1fb0)
![image](https://github.com/zouyaoji/vue-cesium/assets/103714416/b11d0c16-260c-41df-a224-5d9b926d8f99)
我想在before-load（图标点还未加载的时候）中对preRenderDatas参数进行初始化
然后在reday方法中修改图标和名称根据接口（接口是异步的）
![image](https://github.com/zouyaoji/vue-cesium/assets/103714416/28641f0c-5a9c-45ac-ad37-8c8783184a41)
但是2个方法不是我预期的before-load里面的方法执行完成后再执行reday中的方法，最终导致页面图标点无法加载出来

**Comment by zouyaoji** [comment, marked as answer]
你好，可以在你异步请求完成之后再加载组件哈，用v-if。 然后想灵活一点可以参考源码进行初始化对象，
https://github.com/zouyaoji/vue-cesium/blob/dev/packages/composables/use-drawing/use-drawing-polyline.ts#L69

最终都是存到renderDatas里面的。
这儿有些参考代码

https://github.com/kireiyume/cesium-supermap-demo/blob/master/src/views/map/index.vue

**Reply by xiaobaicxk** [reply]
好的，谢谢大佬
